### ISSUE TITLE
Warning log contains newline in a message

### ISSUE THREAD

**Issue by Kokokokoka** [matrix-org/synapse#11350]
### Description
In the log files there's an extra \n which can lead to problems with parsing
```
2021-11-16 01:13:12,111 - synapse.storage.txn - 565 - WARNING - POST-13293904 - [TXN OPERROR] {insert_linearized_receipt-144e005} could not serialize access due to concurrent update
 0/5
```

### Steps to reproduce
1. set loglevel to WARN
2. wait for a concurrent update warning
3. use grep -vr WARNING or anything els
4. find extra lines:
```
./homeserver.log: 0/5
./homeserver.log: 1/5
./homeserver.log: 0/5
./homeserver.log: 1/5
```

### Version information
1.46

- **Install method**:
 repo
- **Platform**:
 debian linux

**Comment by DMRobertson** [comment]
Comes from here:

https://github.com/matrix-org/synapse/blob/44acd8ae0bfffe80e2c2c7aa9e3e0b227d9fb28d/synapse/storage/database.py#L576

**Comment by DMRobertson** [comment]
Comes from here:

 https://github.com/matrix-org/synapse/blob/44acd8ae0bfffe80e2c2c7aa9e3e0b227d9fb28d/synapse/storage/database.py#L590-L607

**Comment by richvdh** [comment]
> `synapse.storage.txn - 565`

I think you'll find it's line 565 :)

**Comment by DMRobertson** [comment]
Whoops, I got the wrong `except` block on develop. Thanks!

**Comment by Kokokokoka** [comment]
Can one strip \n in this message (strip e), or is there a better solution?
Maybe there should be general protection against \n in the log messages as it messes parsing, log collection?

**Comment by richvdh** [comment]
yes, stripping out the newline is probably the best solution here.

Generally, if you want to be parsing logs, I'd recommend considering the structured logging support - we also sometimes log stacktraces which also contain newlines (obviously).
